tensorpack/dataflow/base.py

```diff
@@ -39,29 +40,59 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         return False
 
 
-@six.add_metaclass(ABCMeta)
+class DataFlowMeta(type):
+    """
+    DataFlow uses "__iter__()" and "__len__()" instead of
+    "get_data()" and "size()". This add back-compatibility.
+    """
+    def __new__(meta, name, bases, dct):
+        # "'get_data' in dct" is important as we just care about DataFlow itself
+        # and not derived classes.
+        if '__iter__' not in dct and 'get_data' in dct:
+            dct['__iter__'] = dct['get_data']
+            log_deprecated("DataFlow.get_data()",
```

Comment:
Let's deprecate them after a while. For big changes like this let's first apply it and see what happened.

Reply:
My terminal is flashing anyway because of `python-prctl`. A few more warning would be ok.